py-polars/polars/functions/eager.py

```diff
@@ -231,6 +240,14 @@ def concat(
                 )
             )
         elif how == "horizontal":
+            if strict:
+                nrows = first.select(F.len()).collect()[0, 0]
```

Comment:
The reason this should be implemented on the rust side is that this `collect` here could trigger a massive computation if the query plan is complex, which then gets tossed. The check should be performed when the concatenation operation is actually applied.

Reply:
Understood. When I initially thought about it, I failed to take into account how I would compare the number of rows on Lazyframes.